.github/workflows/kanister-image-build.yaml

```diff
@@ -0,0 +1,39 @@
+name: build-kanister-image
+
+on:
+  pull_request:
+    branches:
+    - master
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: kanisterio/build
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        filters: |
+          modified:
+            - 'docker/build/**'
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+      if: steps.filter.outputs.modified == 'true'
+    - name: Login to Docker Hub
+      uses: docker/login-action@v2
+      if: steps.filter.outputs.modified == 'true'
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Build and push
+      uses: docker/build-push-action@v4
```

Comment:
We should push the image only once the PR is approved and merged.

Reply:
Good point. If the workflow is on demand, then this would work.
Otherwise, we would need to push only on merge, not on the PR stage.

Reply:
made the workflow as on-demand